neural_chat/tests/optimization/test_optimization.py

```diff
@@ -16,7 +16,7 @@
 # limitations under the License.
 
 import unittest
-import os
+from transformers import BitsAndBytesConfig
 from neural_chat.chatbot import build_chatbot
```

Comment:
should we check whether the lib is available or not when used? instead of adding it by default but not used e.g., on CPU for now

Reply:
BitsAndBytesConfig is from transformers, won't be a problem even when bitsandbytes is not avaliable.

I've also added logic to use bitsandbytes only when device is cuda.